lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -262,6 +262,46 @@ ::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::UnsqueezeOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  int32_t dim = getDim();
+
+  // Check that the dim is within the bounds of the input tensor
+  if (dim >= inputType.getRank() || dim < 0) {
```

Comment:
If tensor is `[a, b]`, then we should be able to do:
`[1, a, b]`
`[a, 1, b]`
`[a, b, 1]`

Which means that `dim` can be set to `0`, `1`, or `2`, so it seems that we should allow `dim == inputType.getRank()`?